src/llvm/types/di.rs

```diff
@@ -430,3 +444,136 @@ impl DISubprogram<'_> {
         };
     }
 }
+
+/// Represents the operands for a [`DICompileUnit`]. The enum values correspond
+/// to the operand indices within metadata nodes.
+#[repr(u32)]
+enum DICompileUnitOperand {
+    EnumTypes = 4,
+}
+
+/// Represents the debug information for a compile unit in LLVM IR.
+#[derive(Clone)]
+pub struct DICompileUnit<'ctx> {
+    value_ref: LLVMValueRef,
+    _marker: PhantomData<&'ctx ()>,
+}
+
+impl DICompileUnit<'_> {
+    /// Constructs a new [`DICompileUnit`] from the given `value_ref`.
+    ///
+    /// # Safety
+    ///
+    /// This method assumes that the provided `value_ref` corresponds to a valid
+    /// instance of [LLVM `DICompileUnit`](https://llvm.org/doxygen/classllvm_1_1DICompileUnit.html).
+    /// It's the caller's responsibility to ensure this invariant, as this
+    /// method doesn't perform any valiation checks.
+    pub(crate) unsafe fn from_value_ref(value_ref: LLVMValueRef) -> Self {
+        Self {
+            value_ref,
+            _marker: PhantomData,
+        }
+    }
+
+    pub fn enum_types(&self) -> impl Iterator<Item = DICompositeType> {
+        let llvm_enum_types =
+            unsafe { LLVMGetOperand(self.value_ref, DICompileUnitOperand::EnumTypes as u32) };
+
+        let llvm_enum_types_len = if llvm_enum_types.is_null() {
+            0
+        } else {
+            unsafe { LLVMGetNumOperands(llvm_enum_types) }
+        };
+
+        (0..llvm_enum_types_len).map(move |i| unsafe {
+            let enum_type = LLVMGetOperand(llvm_enum_types, i as u32);
+            DICompositeType::from_value_ref(enum_type)
+        })
+    }
+
+    pub fn replace_enum_types(&mut self, builder: LLVMDIBuilderRef, rep: &[DICompositeType]) {
```

Comment:
```suggestion
    pub fn replace_enum_types(&mut self, builder: LLVMDIBuilderRef, rep: impl IntoIter<Item = DICompositeType>) {
```

or

```suggestion
    pub fn replace_enum_types<I>(&mut self, builder: LLVMDIBuilderRef, rep: I)
    where
        I: IntoIterator<Item = DICompositeType>,
    {
```

So you can pass both slices and lazy iterators here.

Reply:
Done